src/typeInference/annotateTypeSwitchOperator.ts

```diff
@@ -0,0 +1,123 @@
+import {
+	SequenceMultiplicity,
+	SequenceType,
+	stringToSequenceMultiplicity,
+	stringToValueType,
+	ValueType,
+} from '../expressions/dataTypes/Value';
+import astHelper, { IAST } from '../parsing/astHelper';
+
+/**
+ * Checks the type from the argument against all the different cases and if one matches, the return type of that case is returned.
+ * If no types match, the default type is returned.
+ * @param ast the AST to be annotated.
+ * @param argumentType the type of the input.
+ * @param caseClausesReturns the different return types from the cases.
+ * @param defaultCaseReturn the return type of the default case.
+ * @returns the return type of the case that matched.
+ */
+export function annotateTypeSwitchOperator(
+	ast: IAST,
+	argumentType: SequenceType,
+	caseClausesReturns: SequenceType[],
+	defaultCaseReturn: SequenceType
+): SequenceType | undefined {
+	// TODO: check this case in more detail (anyKindTest for example returns undefined)
+	if (!argumentType || caseClausesReturns.includes(undefined)) {
+		return {
+			type: ValueType.ITEM,
+			mult: SequenceMultiplicity.ZERO_OR_MORE,
+		};
+	}
+	// Get the types of the conditions from the clauses
+	const caseClausesConditions = astHelper.getChildren(ast, 'typeswitchExprCaseClause');
+	// Do the switch case to see what will be returned
+	for (let i = 0; i < caseClausesReturns.length; i++) {
+		// The typeswitch either contains a sequenceType with a single type or an 'or' statement (sequenceTypeUnion)
+		const condition: IAST = astHelper.getFirstChild(caseClausesConditions[i], '*');
+		// Check if we have a sequenceType or a sequenceTypeUnion
+		switch (condition[0]) {
+			case 'sequenceType':
+				// Check if the type of the argument matches the type of the condition
+				const result = checkComparison(condition, argumentType, caseClausesReturns[i]);
+				// If the types did not match, check the next case, if they matched, return the type
+				if (!result) continue;
```

Comment:
This is so weird, this should be conflicting with our prettier set-up. I will look at this (and more) after integration since it is an autofixable linting error.

Reply:
That's interesting, I will at least fix the formatting for the time being.